pkg/cloudprovider/aws.go

```diff
@@ -102,11 +104,12 @@ func (p *AWSProvider) UpdateCloudSecret(secret, cloudSecret *kapi.Secret) {
 	}
 }
 
-func (p *AWSProvider) UpdateRegistrySecret(secret, registrySecret *kapi.Secret) {
+func (p *AWSProvider) UpdateRegistrySecret(secret, registrySecret *kapi.Secret) error {
```

Comment:
Maybe I'm missing something obvious here, but why add an `error` return if we never actually return an error?

Reply:
@dymurray needed to satisfy the `Provider` interface which is shared between AWS, GCP, Azure. For Azure, it's possible to encounter errors while updating the registry secret since we have to derive a storageAccountKey from the provided info.

```
type Provider interface {
	SetRole(role string)
	UpdateBSL(location *velero.BackupStorageLocation)
	UpdateVSL(location *velero.VolumeSnapshotLocation)
	UpdateCloudSecret(secret, cloudSecret *kapi.Secret)
	UpdateRegistrySecret(secret, registrySecret *kapi.Secret) error
	UpdateRegistryDC(dc *appsv1.DeploymentConfig, name, dirName string)
	Validate(secret *kapi.Secret) []string
	Test(secret *kapi.Secret) error
}
```

Deriving storageAccountKey from info available on secret (needed for mig-registry config)

https://github.com/fusor/mig-controller/blob/6b6701a854f9229a434a496686534da7cd4a901d/pkg/cloudprovider/azure.go#L82-L86